lib/msf/core/post/windows/file_system.rb

```diff
@@ -292,17 +264,93 @@ def create_symlink(_root, link_name, target_name)
             print_error("Something went wrong while creating the symlink. Return value: NTSTATUS #{error} ()")
             return nil
           end
-          result
+
+          result['LinkHandle']
         ensure
           process.close
         end
 
+        # Create a symbolic link on the file system. This function is a suitable replacement for the `mklink /D` shell
+        # command when *directory* is set to true.
+        #
+        # @param [String] link_name The path at which to create the symbolic link.
+        # @param [String] target_name The path that the new symbolic link targets.
+        # @param [Boolean] directory Whether or not the link target is a directory.
+        #
+        # @return [Boolean] Returns true on success or false   on failure.
+        def create_symlink(link_name, target_name, directory: true)
+          flags = directory ? 'SYMBOLIC_LINK_FLAG_DIRECTORY' : ''
+          result = session.railgun.kernel32.CreateSymbolicLinkW(link_name, target_name, flags)
+          unless result['GetLastError'] == SUCCESS
+            print_error("Error creating the symlink. Windows Error Code: #{result['GetLastError']} - #{result['ErrorMessage']}")
+            return true
```

Comment:
If I understand it correctly, it should returns false on failure.

Reply:
Yeah, you're right when I updated that from nil I used the wrong value. Thanks!